### PR TITLE
Android 11+ storage handling

### PR DESCRIPTION
**Pull request by nirvn**
Expect failures before success...

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 614d2170b97ccc5052fb0c28b7a8382840b46216:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-5372154038f528d77c3aa6f1cd2842f131b44d3c-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-5372154038f528d77c3aa6f1cd2842f131b44d3c-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-5372154038f528d77c3aa6f1cd2842f131b44d3c-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-5372154038f528d77c3aa6f1cd2842f131b44d3c-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for d870374fd51ea9a6a8199325a9081804eb758bdc:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-aa2c428b2658f6433d86b10b22a9a56fa15a8354-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-aa2c428b2658f6433d86b10b22a9a56fa15a8354-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-aa2c428b2658f6433d86b10b22a9a56fa15a8354-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-aa2c428b2658f6433d86b10b22a9a56fa15a8354-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for a40cfcc42c41ff401f4c0f50a1bb7007cf86041b:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-caffbd399e8a3dc80c3e198551c31c54e8505f3f-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-caffbd399e8a3dc80c3e198551c31c54e8505f3f-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-caffbd399e8a3dc80c3e198551c31c54e8505f3f-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-caffbd399e8a3dc80c3e198551c31c54e8505f3f-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 9cd938b8997bd6b4e673d969ac8ff02261f2eb2c:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-4c096bc710068413fde21ad94c561f66f15aba04-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-4c096bc710068413fde21ad94c561f66f15aba04-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-4c096bc710068413fde21ad94c561f66f15aba04-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-4c096bc710068413fde21ad94c561f66f15aba04-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 0b172999484afef9665baf84ec53f60393147f21:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-8d242c66fbf63d1f43b100716b8cfdc978d7ce43-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-8d242c66fbf63d1f43b100716b8cfdc978d7ce43-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-8d242c66fbf63d1f43b100716b8cfdc978d7ce43-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-8d242c66fbf63d1f43b100716b8cfdc978d7ce43-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for d61ba1b6ff27a8d6e692f8cf2f3fbeb351ee71d5:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-c9b7a3ed6b9a1331c95eb8aef9a936e13cd0c109-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-c9b7a3ed6b9a1331c95eb8aef9a936e13cd0c109-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-c9b7a3ed6b9a1331c95eb8aef9a936e13cd0c109-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-c9b7a3ed6b9a1331c95eb8aef9a936e13cd0c109-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for edfec39d1fa2c3ca03880692b326fdecc763feb8:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-7fb7a7a4d97671f43720c78d3f7b171a3a75cb57-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-7fb7a7a4d97671f43720c78d3f7b171a3a75cb57-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-7fb7a7a4d97671f43720c78d3f7b171a3a75cb57-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-7fb7a7a4d97671f43720c78d3f7b171a3a75cb57-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 01e688f53f70d16e66bab49dc621554b1ff22d71:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-f210f5a216fbf1c2d44464b050b7c59050324973-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-f210f5a216fbf1c2d44464b050b7c59050324973-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-f210f5a216fbf1c2d44464b050b7c59050324973-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2298-f210f5a216fbf1c2d44464b050b7c59050324973-x86.apk)

File: platform/android/src/ch/opengis/qfield/QFieldActivity.java [hunk 1]
```diff
@@ -159,4 +174,42 @@ private void checkPermissions()
             ActivityCompat.requestPermissions(QFieldActivity.this, permissions, 101);
         }
     }
+
+    private void checkAllFileAccess()
+    {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R && !Environment.isExternalStorageManager() && !sharedPreferences.getBoolean("DontAskAllFilesPermission", false)) {
+            AlertDialog.Builder builder = new AlertDialog.Builder(this);
+            builder.setTitle("All files access");
+            builder.setMessage(Html.fromHtml("QField needs all files access permission to be able to open local projects and datasets as well as supporting custom projection grids, fonts, and base maps. While not required, <i>allowing access is highly recommended</i>.<br><br>On the next screen, please select <b>QField</b> and select the <b>Allow access to all files</b> option.", Html.FROM_HTML_MODE_LEGACY));
```
Comment (m-kuhn) [inline platform/android/src/ch/opengis/qfield/QFieldActivity.java:183]: Should this come from the xml?

**Review by m-kuhn — APPROVED** [review]

File: platform/android/src/ch/opengis/qfield/QFieldActivity.java (hunk 1 above)
Comment (nirvn) [reply platform/android/src/ch/opengis/qfield/QFieldActivity.java:183]: Yes it should. For some reason it fails though. I must be missing a really basic thing here.

Comment (nirvn) [reply platform/android/src/ch/opengis/qfield/QFieldActivity.java:183]: Will fix in a follow up PR. Let's see if it uploads to store.

**Pull request merged**